adsingestp/parsers/base.py

```diff
@@ -543,6 +543,13 @@ def _detag(self, r, tags_keep):
             for e in elements:
                 if t in self.HTML_TAGS_DANGER:
                     e.decompose()
+                elif (t == "alternatives") or (t == "inline-formula"):
+                    alt_math_element = e.find_all("mml:math", [])
+                    alt_tex_element = e.find_all("tex-math", [])
+                    if alt_math_element and alt_tex_element:
+                        alt_tex_element[0].decompose()
```

Comment:
Is there any case where you might have multiple `tex-math` elements within a single `inline-formula` or `alternatives`?  If so, L550 will leave elements[1:] in `e`.

Reply:
Unlikely, but could there be cases where there are multiple math-element *and* tex-element, *and* there isn't a 1:1 correspondence between the two?  In that case, you'd have to find a way to make sure the tex-element you're about to decompose has a corresponding math-element, probably via an attribute.  I don't have a feel for whether any content types, jats or otherwise, might do that.

Reply:
For 1st point, I will explicitly decompose all tags. 

RE point #2: I haven't seen such a case _within_ an `alternatives` or `inline-formula` tags since they explicitly correspond to a single formula/equation. I thought about this, but the content inside the mml and tex tags are very different so I'm not sure how I could even check them